data-raw/northern-ireland-hsct-secondary-care.R

```diff
@@ -0,0 +1,145 @@
+# Northern Ireland - Secondary Care Indicators
+# Health and Social Care Trusts
+
+# ---- Load libs & helpers ----
+library(tidyverse)
+library(lubridate)
+library(sf)
+library(healthyr)
+library(compositr)
+library(ggridges)
+library(devtools)
+library(readODS)
+library(geographr)
+
+hsct <- boundaries_trusts_ni18 |>
+  st_drop_geometry()
+
+# Population counts
+# Source: https://www.ninis2.nisra.gov.uk/public/AreaProfileReportViewer.aspx?FromAPAddressMulipleRecords=Northern@@Northern@20?
+
+url <- "https://www.ninis2.nisra.gov.uk/Download/Population/Population%20Estimates%20Broad%20Age%20Bands%20(administrative%20geographies).ods"
+
+pop_raw <- download_file(url, ".ods")
+
+pop_ni_hsct <- read_ods(
+  pop_raw,
+  sheet = "HSCT",
+  range = "A4:C9"
+) |>
+  select(
+    trust18_name = "HSCT",
+    trust18_code = "HSCT Code",
+    population = "Persons"
+  )
+
+# ---- RTT ----
+# Higher = worse performance
+rtt <- ni_rtt_hsct |>
+  ungroup() |> # was grouped in ni_rtt_hsct
+  select(
+    trust18_name = "hsct22_name",
+    date,
+    waits_over_18_weeks
+  ) |>
+  filter(date >= max(date) %m-% months(3)) |> # Last 3 months
+  left_join(pop_ni_hsct) |>
+  mutate(percent = waits_over_18_weeks / population) |>
+  group_by(trust18_name) |>
+  summarise(
+    number = sum(waits_over_18_weeks),
+    percent = sum(percent, na.rm = FALSE)
+  ) |>
+  mutate(
+    variable = "Referral to treatment \nwaiting times (Mar 22 - Jun 22)",
+    .after = trust18_name
+  ) |>
+  filter(!trust18_name %in% c("DPC", "Day Case Procedure Centre"))
+
+# ---- Beds ----
+# Higher = better performance
+available_beds <- ni_beds |>
+  left_join(hsct, by = "trust18_code") |>
+  mutate(percent_available = ((total_available_beds - total_occupied_beds) /
+                                total_available_beds)) |>
```

Comment:
Please reformat to this style:

```r
mutate(
    percent_available = ((total_available_beds - total_occupied_beds) /
      total_available_beds)
  ) 
```

Reply:
I'm still not sure what `total_available_beds` referred to. I'm inclined to think it means the number of beds that aren't occupied, rather than the total numbers of beds regardless of their state. As such, you can update you formula to calculate the percentage available as:

```r
percent_available = total_available_beds / (total_available_beds + total_occupied_beds)
```

What do you think?

Reply:
Yes, that makes more sense!